src/types.ts

```diff
@@ -10,7 +10,8 @@ export enum ConnectionState {
 
 export enum FetchConnectionMetadataError {
   /**
-   * Fetch was aborted.
+   * Fetch was aborted. A client can return this if they decided not to fetch
+   * connection metadata _because_ the abort controller was already triggered.
```

Comment:
```suggestion
   * Fetch was aborted. A client must return this if the `AbortSignal` passed is aborted.
   * If the client chooses to abort the request, it must call `signal.abort()` and return this.
```

Is

> If the client chooses to abort, it must call `signal.abort()` and return this.

correct? or does it need to be `client.close()`?

Reply:
Client calls `client.close`... which I have no idea if you can actually safely do inside a fetchConnectionMetadata call.

[AbortSignal instances](https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal) don't have an abort method (the controller has it). I _think_ that confusion is what led to this assertion seeming reasonable in the first place: only Crosis can trigger an abort.

Reply:
The rest of your suggestion feels like it is orthogonal to this change. 

The client _may_ return Aborted now iff we're aborted; it may also return valid metadata that just won't be used. It may _not_ return Aborted if we're not aborted.

This is basically just a convenience handler for valid types.

Reply:
```suggestion
   * Fetch was aborted. A client may return this only if they decided not to fetch
   * connection metadata _because_ the `AbortSignal` was already aborted.
```

Reply:
Maybe this is better.

Reply:
Yeah that seems more sane to me.